### PR TITLE
Add pre-commit to standardize toffy documentation

### LINKED ISSUE #345: Add a pre-commit hook for docstrings
**Is your feature request related to a problem? Please describe.**

The docstrings in `toffy` are inconsistently formatted. It would be nice to include a pre-commit hook that verifies formatting.

**Describe the solution you'd like**

This article lists a few that we could consider: https://towardsdatascience.com/automate-your-python-code-documentation-with-pre-commit-hooks-35c7191949a4. `pydocstyle` looks nice.

### PR DESCRIPTION
**Pull request by camisowers**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Closes #345. Make sure all functions have a docstring and that they're formatted the same throughout the repo.

**How did you implement your changes**
Use `ruff` for docstring formatting, very similar to what @srivarra implemented in [angelolab/maldi-tools](https://github.com/angelolab/maldi-tools).

**Remaining issues**

Decide if we want to enforce any type checking (later). I tested adding it to the pre-commit file and the amount of failures was enough to be it's own PR.

### CODE REVIEW

File: tests/utils/test_utils.py [hunk 1]
```diff
@@ -222,90 +219,110 @@ def mock_visualize_mph(mph_df, out_dir, return_plot=True, regression: bool = Fal
         fig = plt.figure()
         ax1 = fig.add_subplot(111)
         ax1.scatter(mph_df["cum_total_count"], mph_df["MPH"])
-        ax2 = ax1.twiny()
```
Comment (alex-l-kong) [inline tests/utils/test_utils.py:225]: We should leave this in, if I remember correctly the QC visualization needs this.

**Review by alex-l-kong — CHANGES_REQUESTED** [review]
The formatter some weird stuff to the documentation, as well as some lines of code.

File: src/toffy/detector_sweep.py [hunk 2]
```diff
@@ -5,20 +5,22 @@
...
+    """Performs string manipulations to get the sweep parameters from an FOV folder.
 
     Args:
         sweep_name: the name of a folder containing a single FOV of a detector sweep
 
-    Returns
-        tuple: the extracted voltage, date, and time"""
-
+    Returns:
+        tuple: the extracted voltage, date, and time
```
Comment (srivarra) [inline src/toffy/detector_sweep.py:22]: The return type should be the dataclass `VoltageSweepFile`

File: src/toffy/fov_watcher.py [hunk 3]
```diff
@@ -337,7 +336,7 @@ def _process_missed_fovs(self, path: str):
 
         # NOTE: from observation, only the most recent FOV will ever be in danger of timing out
         # so all the FOVs processed in this function should already be fully processed
-        bin_dir = str(Path(path).parents[0])
+        str(Path(path).parents[0])
```
[the remark this answers is not in the document]
Comment (camisowers) [reply src/toffy/fov_watcher.py:339]: Weird that it only removed the assignment and not the whole line. 🙄

File: src/toffy/normalize.py [hunk 4]
```diff
@@ -81,14 +79,14 @@ def write_mph_per_mass(
...
-    """Creates a function of specified type to be used for fitting a curve
+    """Creates a function of specified type to be used for fitting a curve.
 
     Args:
         obj_func (str): the desired objective function. Must be either poly_2, ..., poly_5, or log
 
     Returns:
-        function: the function which will be optimized"""
-
+        function: the function which will be optimized
```
Comment (srivarra) [inline src/toffy/normalize.py:88]: The return type should be `Callable` in cases where a function returns another function.

File: src/toffy/streak_detection.py [hunk 5]
```diff
@@ -169,7 +170,9 @@ def _make_binary_mask(
     return binary_mask
 
 
-def _make_mask_dataframe(streak_data: StreakData, min_length: int = 70) -> None:
+def _make_mask_dataframe(
+    streak_data: StreakData, min_length: int = 70, eccentricity_value=0.9999999
```
[the remark this answers is not in the document]
Comment (srivarra) [reply src/toffy/streak_detection.py:174]: I don't believe it's user adjustable.

[the remark this answers is not in the document]
Comment (camisowers) [reply src/toffy/streak_detection.py:174]: it was in the function, but df.query() wouldn't recognize the use of the variable so the formatter kept removing it 😭

**Review by srivarra — APPROVED** [review]
Just a couple of things extra changes.

**Review by srivarra — CHANGES_REQUESTED** [review]
Just a couple of things extra changes.

File: tests/utils/test_utils.py (hunk 1 above)
Comment (camisowers) [reply tests/utils/test_utils.py:225]: It's for mph and this is just a mock plotting for watcher testing. None of the plot ticks are correctly set anyways.

**Review by alex-l-kong — APPROVED** [review]

**Review by srivarra — APPROVED** [review]
Looks good!

**Review by ngreenwald — APPROVED** [review]
Fun stuff, thanks for doing this

**Pull request merged**